libs/framework/src/bundle_archive_private.h

```diff
@@ -23,13 +23,57 @@
 
 #include "bundle_archive.h"
 
-celix_status_t bundleArchive_create(celix_framework_t* fw, const char *archiveRoot, long id, const char *location, const char *inputFile,
-                                    bundle_archive_pt *bundle_archive);
+#ifdef __cplusplus
+extern "C" {
+#endif
 
-celix_status_t bundleArchive_createSystemBundleArchive(celix_framework_t* fw, bundle_archive_pt *bundle_archive);
+#define CELIX_BUNDLE_ARCHIVE_STATE_PROPERTIES_FILE_NAME "bundle_state.properties"
 
-celix_status_t bundleArchive_recreate(celix_framework_t* fw, const char *archiveRoot, bundle_archive_pt *bundle_archive);
+#define CELIX_BUNDLE_ARCHIVE_SYMBOLIC_NAME_PROPERTY_NAME "bundle.symbolic_name"
+#define CELIX_BUNDLE_ARCHIVE_VERSION_PROPERTY_NAME "bundle.version"
+#define CELIX_BUNDLE_ARCHIVE_BUNDLE_ID_PROPERTY_NAME "bundle.id"
+#define CELIX_BUNDLE_ARCHIVE_LOCATION_PROPERTY_NAME "bundle.location"
+
+#define CELIX_BUNDLE_ARCHIVE_RESOURCE_CACHE_NAME "resources"
+#define CELIX_BUNDLE_ARCHIVE_STORE_DIRECTORY_NAME "storage"
+
+#define CELIX_BUNDLE_MANIFEST_REL_PATH "META-INF/MANIFEST.MF"
+
+/**
+ * @brief Create bundle archive.
+ *
+ * Takes ownership of archiveRoot.
```

Comment:
```suggestion
 *
```
`bundleArchive_create` duplicates `archiveRoot` rather than takes ownership.